IGHTMLQuery/Classes/IGXMLNodeQuery.h

```diff
@@ -19,7 +19,7 @@
  @param xpath used to query the document
  @return elements matched by supplied XPath query.
  */
-- (IGXMLNodeSet*) queryWithXPath:(NSString*)xpath;
+- (nullable IGXMLNodeSet*) queryWithXPath:(nullable NSString*)xpath;
```

Comment:
As you pointed out, if we cannot alloc a new object, the program will not work anyway. 

We need to check init returns if the super class has override the init, in NSObject case, init do nothing and we should not worry about it. (NSObject initializer is also not faillable in Swift)